src/webgpu/api/operation/memory_sync/buffer/buffer_sync_test.ts

```diff
@@ -0,0 +1,242 @@
+import { assert } from '../../../../../common/framework/util/util.js';
+import { GPUTest } from '../../../../gpu_test.js';
+
+const SIZE = 4;
+export class BufferSyncTest extends GPUTest {
+  // Create a buffer, and initiate it to a specified value for all elements.
+  async createBufferWithValue(initValue: number): Promise<GPUBuffer> {
+    const fence = this.queue.createFence();
+    this.queue.signal(fence, 1);
+    const data = new Uint32Array(SIZE / 4);
+    for (let i = 0; i < SIZE / 4; ++i) {
+      data[i] = initValue;
+    }
+    const buffer = this.device.createBuffer({
+      mappedAtCreation: true,
+      size: SIZE,
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST | GPUBufferUsage.STORAGE,
+    });
+    new Uint8Array(buffer.getMappedRange()).set(data);
+    buffer.unmap();
+    await fence.onCompletion(1);
+    return buffer;
+  }
+
+  createBindGroup(
+    pipeline: GPURenderPipeline | GPUComputePipeline,
+    buffer: GPUBuffer
+  ): GPUBindGroup {
+    return this.device.createBindGroup({
+      layout: pipeline.getBindGroupLayout(0),
+      entries: [{ binding: 0, resource: { buffer } }],
+    });
+  }
+
+  // Create a compute pipeline and write given data into storage buffer.
+  createStorageWriteComputePipeline(value: number): GPUComputePipeline {
+    const wgslCompute = `
+      type Data = [[block]] struct {
+        [[offset(0)]] a : i32;
+      };
+
+      [[binding(0), set(0)]] var<storage_buffer> data : Data;
+      [[stage(compute)]] fn main() -> void {
+        data.a = ${value};
+        return;
+      }
+    `;
+
+    return this.device.createComputePipeline({
+      computeStage: {
+        module: this.device.createShaderModule({
+          code: wgslCompute,
+        }),
+        entryPoint: 'main',
+      },
+    });
+  }
+
+  // Create a render pipeline and write given data into storage buffer at fragment stage.
+  createStorageWriteRenderPipeline(value: number): GPURenderPipeline {
+    const wgslVertex = `
```

Comment:
nit: I think both entrypoints could be in the same module probably?

Unless it triggers a validation error, which would be because we don't properly reflect WGSL bindings yet.

Reply:
I tried, but it failed. I did notice that we use one single module for vertex code and fragment code.

Reply:
How did it fail? Is it due to incompleteness in Tint/Dawn?

Reply:
If failed because my local Chromium was too old. It can success now. Done!